langchain4j-mcp/src/main/java/dev/langchain4j/mcp/client/DefaultMcpClient.java

```diff
@@ -0,0 +1,149 @@
+package dev.langchain4j.mcp.client;
+
+import static dev.langchain4j.internal.Utils.getOrDefault;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotNull;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import dev.langchain4j.agent.tool.ToolExecutionRequest;
+import dev.langchain4j.agent.tool.ToolSpecification;
+import dev.langchain4j.mcp.client.protocol.InitializeParams;
+import dev.langchain4j.mcp.client.protocol.McpCallToolRequest;
+import dev.langchain4j.mcp.client.protocol.McpInitializeRequest;
+import dev.langchain4j.mcp.client.protocol.McpListToolsRequest;
+import dev.langchain4j.mcp.client.transport.McpTransport;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// TODO: currently we request a new list of tools every time, so we should
+// add support for the `ToolListChangedNotification` message, and then we can
+// cache the list
+public class DefaultMcpClient implements McpClient {
+
+    private static final Logger log = LoggerFactory.getLogger(DefaultMcpClient.class);
+    private final AtomicLong idGenerator = new AtomicLong(0);
+    private final McpTransport transport;
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private final String clientName;
+    private final String clientVersion;
+    private final String protocolVersion;
+
+    public DefaultMcpClient(Builder builder) {
+        transport = ensureNotNull(builder.transport, "transport must not be null");
```

Comment:
```suggestion
        transport = ensureNotNull(builder.transport, "transport");
```
nit, "cannot be null" will be added automatically